src/App/Multisite.php

```diff
@@ -135,4 +125,35 @@ public function getClientUrl($host = null)
 
         return $this->subdomain . '.' . getenv('MULTISITE_CLIENT_DOMAIN');
     }
+    protected function checkDeploymentStatus($deployment)
+    {
+        // No deployment? throw a 404
+        if (! count($deployment)) {
+            abort(404, "Deployment not found");
+        }
+
+        $status = $deployment['status'];
+        $deployed_date = $deployment['deployed_date'];
+
+        if ($status === 'migrating' && !$deployedDate) {
+            abort(503, "Deployment not ready");
+        }
+        if (($status === 'migrating' && $deployedDate) || $status === 'maintenance') {
+            abort(503, $this->domain . "is down for maintenance");
+        }
+        if ($status === 'pending') {
+            abort(503, $this->domain . "is not ready");
+        }
```

Comment:
We could combine this into the one above as an else if, I don't know if that's better or not but at least it groups things on the basis of status checks